SymPEP-0001.md

```diff
@@ -0,0 +1,241 @@
+the [SymPEP repository](https://github.com/sympy/SymPEPs) on GitHub.
+
+Not all significant changes or additions to SymPy require a SymPEP. If the
+change would benefit from extended discussion or needs a roadmap for
+implementation, a SymPEP should be considered. If unsure, consult with the
+community on the currently active discussion forum. Most all changes to and
+decisions regarding SymPy can be made through issues and pull requests, so it
+is best to start with one or the other. Examples of things that likely need a
+SymPEP are: major version changes (1.X.X -> 2.0), large breaks in backwards
+compatibility, adding new primary packages within SymPy, adding hard
+dependencies, introduction of development practices or tools all developers
+need to adopt, etc. Long term SymPy policies and goals should also be discussed
+and developed as SymPEPs. Statements of these policies and goals should be
+incorporated into the SymPy documentation, as determined in the associated
+SymPEP.
+
+Every SymPEP must have at least one champion. The champion(s) are persons who
+are responsible for writing the SymPEP, leading the discussions around it, and
+developing consensus around it.
+
+The author of a SymPEP should fork the repository and create a draft pull
+request with a new SymPEP document based on the [SymPEP
+template](SymPEP-template). The SymPEP document may be named `SymPEP-XXXX.md`
+until a number is assigned. Once the author completes all information in the
+template and would like it to be reviewed, one of the core SymPy developers should
+decide if the proposal is a legtimate proposal and then assign a number to the
+SymPEP, in which case `XXXX` should be replaced with the number with leading
+0s. The pull request can then be moved to an open pull request and is ready for
+community feedback and discussion but remains in draft state. The person who
+assigns the number should also update the
+[README](https://github.com/sympy/SymPEPs/blob/main/README.md) of the main
+[SymPEP repository](https://github.com/sympy/SymPEPs) to list that number.
+Numbers should be assigned to SymPEPs as soon as it is determined that they are
+a legitimate proposal. If a SymPEP is not deemed necessary, the author should
+rework the SymPEP or move it to a normal pull request or issue on the main
+SymPy repository. If a SymPEP ends up being rejected or postponed, it keeps its
+number, as rejection or postponement status is still a discussion that should
+be documented in the SymPEP. SymPEP numbers should generally be assigned in
+increasing numeric order.
+
+Discussion on the SymPEP should then continue on the open pull request.
+Discussions may also take place in other places, such as [GitHub
+discussions](https://github.com/sympy/SymPEPs/discussions) or the [mailing
+list](http://groups.google.com/group/sympy). All discussions should be
+cross-referenced in the "Discussions" section of the SymPEP document.
```

Comment:
> One forum (I suggest the mailing list) should at least be designated as the primary place in which information about SymPEPs is announced.

I don’t like the mailing list because it isn’t very readable, and the conversation is linear, which means people can start ignoring some comments in the middle or jump into different topics, contexts all the time.

I have the people in my work doing almost all the conversation inside github. 
It’s a preference, however, I don’t find a good reason to strongly prefer mailing list other than the reason of accessibility.

Reply:
> I don’t like the mailing list because it isn’t very readable

The first SymPEP that I intend to propose will be to migrate the mailing list to Discourse.

Reply:
Okay, that may improve the experience of using mailing list. But it seems like I'd less prioritize opening a discussion threads about my SymPEPs, until this is done or the mailing list is improved.

Reply:
The point is that there needs to be a place where there are more people than those who are subscribed to GitHub notifications. The mailing list has 4000 subscribers which is massive compared to the number of people involved in any GitHub discussion. Also some people prefer not to use GitHub and should be able to participate in some way without needing a GitHub account.

Reply:
@sylee957 are you making an objection to the draft SymPEP I need to address or just making a passing comment?

Reply:
I may approve

Reply:
One way to handle this is to merge the SymPEP PR as soon as possible, so that it's just there to get things in the repo. This is how NEPs work. https://numpy.org/neps/nep-0000.html

> At the earliest convenience, the PR should be merged (regardless of whether it is accepted during discussion). Additional PRs may be made by the Author to update or expand the NEP, or by maintainers to set its status, discussion URL, etc.